File: Sorting Algorithms/GO/bubble_sort.go
package main

import "fmt"

func bubbleSort(numbers []int) []int {
	if numbers != nil && len(numbers) > 0 {
		for i := 0; i < len(numbers); i++ {
			for j := 0; j < len(numbers); j++ {
				if numbers[i] < numbers[j] {
					temp := numbers[i]
					numbers[i] = numbers[j]
					numbers[j] = temp
				}
			}
		}
	} else {
		fmt.Println("Empty or Null Array")
	}
	return numbers
}

// main function
func main() {

	fmt.Println("Bubble Sort \n")

	// creating an integer array with values
	numbers := []int{4, 2, 1, 5, 3}

	fmt.Println("Before Sorting:", numbers)

	// calling the bubbleSort function
	numbers = bubbleSort(numbers)

	fmt.Println("\nAfter Sorting:", numbers)
}

File: Sorting Algorithms/Dart/BubbleSort.dart

bubbleSort(List numbers) {

        for (int i = 1; i < numbers.length; i++) {
                for (int j = 0; j < numbers.length - i; j++) {
                        if (numbers[j] > numbers[j + 1]) {
                                int temp = numbers[j];
                                numbers[j] = numbers[j + 1];
                                numbers[j + 1] = temp;
                        }
                }
        }

}


displayNumbers(List numbers) {
        for (int i = 0; i < numbers.length; i++) {
                print(numbers[i]);
        }
}



void main() {

        var numbers = [10,1,4,2,6];
        bubbleSort(numbers);
        displayNumbers(numbers);

}

